### PR TITLE
Update azure storage account authentication documentation

### LINKED ISSUE #4658: Document that `az login` identity also works for Azure
https://restic.readthedocs.io/en/stable/030_preparing_a_new_repo.html#microsoft-azure-blob-storage

For example, the following works if the user has the `Storage Blob Data Contributor` role on the storage account:

```sh
export AZURE_RESOURCE_GROUP="some-group"
export AZURE_ACCOUNT_NAME="someaccount"
az login
restic -r azure:backup:/ init
```

This can be run on any machine, not just Azure VMs.

### PR DESCRIPTION
**Pull request by konidev20**
What does this PR change? What problem does it solve?
-----------------------------------------------------
Restic documentation now contains information about using `az login` CLI and 
the required minimum role assignments to authenticate to Azure Blob Storage 
Accounts.

Added a section in the documentation to show how to set the `AZURE_RESOURCE_GROUP` and the 
`az login` command before initialising a repository into the Azure blob storage account. Also, recommended
the minimum role assignment of `Storage Blob Data Contributor` to use the `az login` capability. 

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Closes #4658 

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- ~[ ] I have added tests for all code changes.~
- - This was a documentation change
- [x] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

File: changelog/unreleased/issue-4658 [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Change: Updated Azure Blob Storage Authentication documentation to use `az login`
```
Comment (MichaelEischer) [inline changelog/unreleased/issue-4658:1]: We usually don't write changelog entries for documentation updates, so please remove the file again, sorry.

**Review by MichaelEischer — COMMENTED** [review]
LGTM, except for the changelog entry that must be removed. Please also squash the commits.

File: changelog/unreleased/issue-4658 (hunk 1 above)
Comment (konidev20) [reply changelog/unreleased/issue-4658]: Hey @MichaelEischer, 

I have removed the change log file and also squashed all my commits into a single one.

**Review by MichaelEischer — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**